### PR TITLE
add test, package, ..

### PR DESCRIPTION
**Pull request by caub**
@jarek-foksa Hello, can you check if this pull request is ok? I'd like to add more tests and use it

I didn't change the code (except the end of arcToCurve to make it more like the raphael code), just reorganized

I'll try to integrate it in jsdom and svgo

### CODE REVIEW

**Comment by jarek-foksa** [comment]
Thanks, I will review it in few days as I'm currently busy working on another project.

File: shim.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      this[$cachedPathData] = null;
+      this[$cachedNormalizedPathData] = null;
+    }
+
+    removeAttribute.call(this, name);
+  };
+
+  SVGPathElement.prototype.getPathData = function(options) {
+    if (options && options.normalize) {
+      if (!this[$cachedNormalizedPathData]) {
```
Comment (caub) [inline shim.js:41]: I noticed how you double-cached the normalized path, but I don't think it's worth

**Comment by jarek-foksa** [comment]
Why do you think it is not worth it? It should be always faster than parsing the path data on each call.

Please put on hold any work on this for now, I'm still not sure whether I want to rely on the whole Babel toolchain and a custom module loader. Considering the project size, I think we might be better off with plain ES5.

**Comment by caub** [comment]
mostly because I tried to put `normalize` as an argument of getPathData, to make it more simple

I mean not worth in terms of perf, since it's also just a polyfill

anyway, are you ok that I take your work, and put it in another repo (well I'll keep it in the fork, it'll be clearer), because I'd rather publish it on npm too

**Comment by jarek-foksa** [comment]
`normalize` is already an argument of `getPathData()`, so I'm not sure what you mean. I guess you are referring to some code in your updated version, I will review it and give you more feedback later this week.

The project is licensed under MIT, so you can do anything you want with it. Just please use a recognisably different project name if you are going to submit it to NPM.

**Comment by caub** [comment]
Sorry, yes, the main idea is above all to test

With your last commit if I do this

```js
require('../path-data-polyfill');

const str = pd => pd.map(seg => seg.type + (seg.values || []).join(" ")).join("");

const path = JSDOM.fragment`<svg>
  <path d="M10,10h10l20-20L10,20a2.2,-6.3,36,1,0,85.3,67.3Z"></path>
</svg>`.firstChild.querySelector('path');

console.log(str(path.getPathData({normalize:true})))
```

and draw it: https://jsfiddle.net/crl/tvy0Ldm4/

it seems incorrect

It's still the same with my changes

**Comment by jarek-foksa** [comment]
The incorrect results seem to be caused by a bug in `arcToCubicCurves` function which has been ported from Raphael.js long time ago. I suspect it might be already fixed upstream (check https://github.com/DmitryBaranovskiy/raphael/blob/master/raphael.js#L1894).

**Comment by caub** [comment]
Yes I saw it, I just made a new commit, and pasted exactly their function https://github.com/caub/path-data-polyfill.js/blob/test/polyfill.js#L311 except their memoizer for rotate and math is Math

still same error, so it's weird

**Comment by jarek-foksa** [comment]
In your example the `ry` parameter of the arc segment has negative value which is invalid / out of range. According to [the spec](https://svgwg.org/svg2-draft/implnote.html#ArcOutOfRangeParameters) we should run `rx` and `ry` throw `Math.abs()` before we do any further computations.

**Comment by jarek-foksa** [comment]
The issue with arcs should be fixed now. After having reviewed your changes, I have chosen not to merge them because they are introducing too many dependencies. I have added package.json, .gitignore and test.html and removed all ES6 code, please let me know if there is anything else that should be backported from your branch.

**Comment by caub** [comment]
Ok thanks good, I'll check tests

Any idea what was the issue? it's hard to tell from the diff, and I see you kept the old arcToCurve

https://github.com/jarek-foksa/path-data-polyfill.js/blob/master/path-data-polyfill.js#L899 can be out of the loop

also https://github.com/jarek-foksa/path-data-polyfill.js/blob/master/path-data-polyfill.js#L483-L502

could be simplified to:

```js
params = [m2, m3, m4].concat(params);

var curves = [];

for (var i = 0; i < params.length; i+=3) {
  var r1 = rotate(params[3*i][0], params[3*i][1], angleRad);
  var r2 = rotate(params[3*i+1][0], params[3*i+1][1], angleRad);
  var r3 = rotate(params[3*i+2][0], params[3*i+2][1], angleRad);
  curves.push([r1.x, r1.y, r2.x, r2.y, r3.x, r3.y]);
}

return curves;
```
and the absolutize can be shrunk a lot too, but I don't think you'll care

**Comment by jarek-foksa** [comment]
> Any idea what was the issue? it's hard to tell from the diff, and I see you kept the old arcToCurve

To resolve the issue I just had to add Math.abs() [here](https://github.com/jarek-foksa/path-data-polyfill.js/blob/master/path-data-polyfill.js#L870-L871).

> https://github.com/jarek-foksa/path-data-polyfill.js/blob/master/path-data-polyfill.js#L899 can be out of the loop (...) also https://github.com/jarek-foksa/path-data-polyfill.js/blob/master/path-data-polyfill.js#L483-L502 could be simplified (...)

Thanks, I have backported those changes (I didn't notice them in the diff because you have renamed the whole file)

> and the absolutize can be shrunk a lot too, but I don't think you'll care

I would prefer to keep the existing more verbose implementation as it feels more readable to me.

**Comment by caub** [comment]
Oops all the [`3*`](https://github.com/jarek-foksa/path-data-polyfill.js/blob/master/path-data-polyfill.js#L488-L490) shouldn't be there (the tests was still passing tho)

**Pull request closed without merging**